src/classes/User.js

```diff
@@ -0,0 +1,66 @@
+// const { recipeData } = require("../data/recipes");
+const { ingredientsData } = require("../data/ingredients");
+// import Recipe from '../src/classes/Recipe';
+
+
+class User {
+  constructor(human) {
+    this.name = human.name;
+    this.id = human.id;
+    this.pantry = human.pantry;
+    this.favoriteRecipes = [];
+    this.recipesToCook = [];
+    this.recipeMatch = [];
+  }
+
+  addFavoriteRecipe(recipe) {
+    if(!this.favoriteRecipes.includes(recipe)) {
+      this.favoriteRecipes.push(recipe)
+    }
+  }
+
+  addRecipeToCook(recipe) {
+    if(!this.recipesToCook.includes(recipe)) {
+      this.recipesToCook.push(recipe)
+    }
+  }
+
+  filterFavoritesByName(userInput) {
+    let recipesByName = this.favoriteRecipes.filter((recipe) => {
+      if(recipe.name.includes(userInput)) {
+        return recipe
+      }
+    })
+    return recipesByName;
+  }
+
+  filterFavoritesByIngredient(userInput) {
```

Comment:
I think we're going to have to create a new view, where we can see this new info pulled up on the DOM.